### PR TITLE
fix IllegalArgumentE when encode for edge

### PR DESCRIPTION
**Pull request by Nicole00**

### CODE REVIEW

**Comment by Nicole00** [comment]
CI failed because of docker-compose's branch change.
re-run this pr when #237 is merged.

File: client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java [hunk 1]
```diff
@@ -213,21 +213,18 @@ public SchemaProviderImpl genSchemaProvider(long ver, Schema schema) {
...
-        TagItem tag = metaCache.getTag(spaceName, schemaName);
         Schema schema;
         long ver;
-        if (tag == null) {
+        try {
+            TagItem tag = metaCache.getTag(spaceName, schemaName);
+            schema = tag.getSchema();
+            ver = tag.getVersion();
+        } catch (IllegalArgumentException e) {
             EdgeItem edge = metaCache.getEdge(spaceName, schemaName);
```
Comment (laura-ding) [inline client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java:223]: here also can throw exception

Comment (Nicole00) [reply client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java:223]: Then exception will be thrown.
If IllegalArguementException is thrown, schemaName is not exist.

Comment (laura-ding) [reply client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java:223]: better unify handle the exception, the exception here should no return the edge exception.here return the error msg is no only edge.

Comment (Nicole00) [reply client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java:223]: OK. Catch getEdge operation for IllegalArgumentException, and thrown new IllegalArgumentException with "schemaName does not exist" msg.

**Review by laura-ding — APPROVED** [review]

**Pull request merged**